document/FileFormat.mdk

```diff
@@ -2408,23 +2412,132 @@ This allows tools to include information about the stack frame that is not expli
+This enables users to easily refer to the location in conversation, for example,
+"I think the problem occurs at location 6".
+~
+
+~ IsoNumberedNote
+The `message` property (&sect;[#annotatedCodeLocation-message]) of an `annotatedCodeLocation` can use the `id` property
+to refer to another `annotatedCodeLocation` within the same `codeFlow` object.
+~
+
+~ IsoExample
+
+```
+    {                                            # A codeFlow object
+        "locations": [
+            {
+                "id": "1",
+                "kind": "call",
+                "physicalLocation": { ... }
+            },
+            {
+                "kind: "callReturn",
+                "physicalLocation": { ... },
+                "message": "Return from call at location 1"
+            }
+        ]
+    }
+```
+~
+
 ### `physicalLocation` property { #annotatedCodeLocation-physicalLocation }
 
 An `annotatedCodeLocation` object shall contain a property named `physicalLocation` whose value is a `physicalLocation` object
 (&sect;[#physicalLocation]) that specifies the file location to which the `annotatedCodeLocation` object refers.
 
+### `fullyQualifiedLogicalName` property { #annotatedCodeLocation-fullyQualifiedLogicalName }
+
+An `annotatedCodeLocation` object may contain a property named `fullyQualifiedLogicalName` whose value
+is a string containing the fully qualified name of the method to which this `annotatedCodeLocation` refers.
+See &sect;[#location-fullyQualifiedLogicalName] for examples.
+
+If the `run.logicalLocations` property (&sect;[#run-logicalLocations]) is present,
+the value of the `fullyQualifiedLogicalName` property should be equal to the name of one of the properties on the `run.logicalLocations` object,
+with one exception, described in &sect;[#annotatedCodeLocation-logicalLocationKey].
+
+### `logicalLocationKey` property { #annotatedCodeLocation-logicalLocationKey }
+
+An `annotatedCodeLocation` object may contain a property named `logicalLocationKey` whose value is a string.
+If present, this string shall be equal to the name of one of the properties on the `run.logicalLocations` object (&sect;[#run-logicalLocations]),
+which provides additional information about the logical location specified by `fullyQualifiedLogicalName` (&sect;[#stackFrame-fullyQualifiedLogicalName]).
+For more information about the purpose of this property, see &sect;[#location-fullyQualifiedLogicalName].
+
+### `module` property { #annotatedCodeLocation-module }
+
+An `annotatedCodeLocation` object may contain a property named `module` whose value is a string
+containing the name of the module that contains the code location specified by this object.
+
+### `threadId` property { #annotatedCodeLocation-threadId }
+
+An `annotatedCodeLocation` object may contain a property named `threadId` whose value is an integer which
+identifies that thread that was executing when the execution of a code flow reached the location specified by this object.
```

Comment:
fixed

---
In reply to: [64918878](https://github.com/sarif-standard/sarif-spec/pull/222#discussion_r64918878) [](ancestors = 64918878)